tests/baselines/reference/user/TypeScript-Node-Starter.log

```diff
@@ -0,0 +1,9 @@
+Exit Code: 2
+Standard output:
+node_modules/mongoose/index.d.ts(438,37): error TS2589: Type instantiation is excessively deep and possibly infinite.
+node_modules/mongoose/index.d.ts(1194,5): error TS2589: Type instantiation is excessively deep and possibly infinite.
+node_modules/mongoose/index.d.ts(1195,5): error TS2589: Type instantiation is excessively deep and possibly infinite.
```

Comment:
I'm pretty sure we don't want mongoose having an error like this - I imagine we need to look into this.

Reply:
Error is on a `Query<any, this>` inside the `Document` class, where `Query<ResultType, DocType extends Document>`. This is a _really simple_ type and absolutely shouldn't infini-stack. The other two are on `Schema` instantiations (which is also just a class with a type argument); I think the commonality is the `FilterQuery` type usage within both; need to investigate this one more; making a minimal repro will take a bit.